### PR TITLE
gNMI-1.4: Added deviation for fanOperStatusUnsupported

### PR DESCRIPTION
**Pull request by stthummuri**
Added deviation for fanOperStatusUnsupported cases where device does not support oper-status leaf for some of the fan components

### CODE REVIEW

File: feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go [hunk 1]
```diff
@@ -588,10 +588,14 @@ func ValidateComponentState(t *testing.T, dut *ondatra.DUTDevice, cards []string
 		}
 
 		if p.operStatus != "" {
-			operStatus := gnmi.Get(t, dut, component.OperStatus().State()).String()
-			t.Logf("Hardware card %s OperStatus: %s", card, operStatus)
-			if operStatus != activeStatus {
-				t.Errorf("component.OperStatus().Get(t) for %q): got %v, want %v", card, operStatus, p.operStatus)
+			if deviations.FanOperStatusUnsupported(ondatra.DUT(t, "dut")) && regexp.MustCompile("Fan").Match([]byte(card)) {
```
Comment (greg-dennis) [inline feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go:591]: You already have the dut, so no need to look it up. Just "deviations.FanOperStatusUnsupported(dut)"

Comment (greg-dennis) [inline feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go:591]: regex is a bit overkill here. Just use strings.HasPrefix or strings.Contains

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go (hunk 1 above)
Comment (stthummuri) [reply feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go]: Done.

Comment (stthummuri) [reply feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go]: Done.

**Review by greg-dennis — DISMISSED** [review]
Approving for Go quality

**Review by greg-dennis — DISMISSED** [review]
Re-approving for Go quality

File: feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go [hunk 2]
```diff
@@ -588,10 +588,14 @@ func ValidateComponentState(t *testing.T, dut *ondatra.DUTDevice, cards []string
 		}
 
 		if p.operStatus != "" {
-			operStatus := gnmi.Get(t, dut, component.OperStatus().State()).String()
-			t.Logf("Hardware card %s OperStatus: %s", card, operStatus)
-			if operStatus != activeStatus {
-				t.Errorf("component.OperStatus().Get(t) for %q): got %v, want %v", card, operStatus, p.operStatus)
+			if deviations.FanOperStatusUnsupported(dut) && strings.Contains(card, "Fan") {
+				t.Logf("Skipping check for fan oper-status")
```
Comment (mojiiba) [inline feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go:592]: please make skip message specific `t.Logf("Skipping check for fan oper-status due to deviation FanOperStatusUnsupported")`

Comment (sgirishj) [reply feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go]: Done. Thanks!

Comment (sgirishj) [reply feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go]: @mojiiba, could you please approve the changes. Thanks!

**Review by sachendras — APPROVED** [review]

**Pull request merged**